### PR TITLE
PM-12782: Handle email verification feature flag when navigating from the intro carousel

### PR DESCRIPTION
**Pull request by matt-livefront**
## 🎟️ Tracking

[PM-12782](https://bitwarden.atlassian.net/browse/PM-12782)

## 📔 Objective

If email verification is enabled, the intro carousel will open the start registration view instead of the previous create account flow.

## 📸 Screenshots

https://github.com/user-attachments/assets/4fae9abf-3878-4304-9d3c-3c9cc41f0915

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

[PM-12782]: https://bitwarden.atlassian.net/browse/PM-12782?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: BitwardenShared/UI/Auth/IntroCarousel/IntroCarouselProcessor.swift [hunk 1]
```diff
@@ -22,22 +29,45 @@ class IntroCarouselProcessor: StateProcessor<IntroCarouselState, IntroCarouselAc
...
         case .logIn:
             coordinator.navigate(to: .landing)
         }
     }
 }
+
+// MARK: - StartRegistrationDelegate
+
+extension IntroCarouselProcessor: StartRegistrationDelegate {
+    func didChangeRegion() async {
```
Comment (phil-livefront) [inline BitwardenShared/UI/Auth/IntroCarousel/IntroCarouselProcessor.swift:70]: is this needed then if theres no-op?

**Review by phil-livefront — APPROVED** [review]

File: BitwardenShared/UI/Auth/IntroCarousel/IntroCarouselProcessor.swift (hunk 1 above)
Comment (matt-livefront) [reply BitwardenShared/UI/Auth/IntroCarousel/IntroCarouselProcessor.swift:70]: I was getting an error if I didn't specify a delegate:

https://github.com/bitwarden/ios/blob/52283cf9be95429ea33d6f4f02df89397a6be78a/BitwardenShared/UI/Auth/AuthCoordinator.swift#L736-L738

I guess this is better than failing silently if you needed the delegate and didn't set it?

File: BitwardenShared/UI/Auth/IntroCarousel/IntroCarouselProcessor.swift [hunk 2]
```diff
@@ -22,22 +29,45 @@ class IntroCarouselProcessor: StateProcessor<IntroCarouselState, IntroCarouselAc
...
 
+    override func perform(_ effect: IntroCarouselEffect) async {
+        switch effect {
+        case .createAccount:
+            let isEmailVerificationEnabled: Bool = await services.configService.getFeatureFlag(
+                .emailVerification,
+                isPreAuth: true
+            )
+            if isEmailVerificationEnabled {
+                coordinator.navigate(to: .startRegistration, context: self)
```
Comment (phil-livefront) [inline BitwardenShared/UI/Auth/IntroCarousel/IntroCarouselProcessor.swift:50]: ❓ should we also be checking the `native-create-account-flow` flag or is that only required if you come into the flow from the log in screen?

File: BitwardenShared/UI/Auth/IntroCarousel/IntroCarouselProcessor.swift (hunk 1 above)
Comment (phil-livefront) [reply BitwardenShared/UI/Auth/IntroCarousel/IntroCarouselProcessor.swift:70]: yeah, this makes sense to me.  thanks for explaining!

File: BitwardenShared/UI/Auth/IntroCarousel/IntroCarouselProcessor.swift (hunk 2 above)
Comment (phil-livefront) [reply BitwardenShared/UI/Auth/IntroCarousel/IntroCarouselProcessor.swift:50]: oh yeah, thats right... my b!

**Pull request merged**